File: source/gui/helper/GUIFonts.java
package gui.helper;

import imgui.ImFont;
import imgui.ImFontAtlas;
import imgui.ImGui;
import imgui.flag.ImGuiFreeTypeBuilderFlags;

public final class GUIFonts {
    private GUIFonts() {}

    public static ImFont loadFont(String filepath, int size) {
        ImFontAtlas fontAtlas = getGlobalAtlas();
        ImFont font = fontAtlas.addFontFromFileTTF(filepath, size, fontAtlas.getGlyphRangesDefault());
        fontAtlas.setFlags(ImGuiFreeTypeBuilderFlags.LightHinting);
        fontAtlas.build();
        return font;
    }

    public static ImFontAtlas getGlobalAtlas() {
        return ImGui.getIO().getFonts();
    }

    public static void clearGlobalAtlas() {
        getGlobalAtlas().clear();
    }

    public static void buildGlobalAtlas() {
        getGlobalAtlas().build();
    }
}

File: source/gui/helper/GUIFonts.kt
package gui.helper

import imgui.ImFont
import imgui.ImFontAtlas
import imgui.ImGui
import imgui.flag.ImGuiFreeTypeBuilderFlags

object GUIFonts {
    fun loadFont(filepath: String?, size: Int): ImFont {
        val fontAtlas = globalAtlas
        val font = fontAtlas.addFontFromFileTTF(filepath, size.toFloat(), fontAtlas.glyphRangesDefault)
        fontAtlas.flags = ImGuiFreeTypeBuilderFlags.LightHinting
        fontAtlas.build()
        return font
    }

    val globalAtlas: ImFontAtlas
        get() = ImGui.getIO().fonts

    fun clearGlobalAtlas() {
        globalAtlas.clear()
    }

    fun buildGlobalAtlas() {
        globalAtlas.build()
    }
}